backend/archive_data.py

```diff
@@ -0,0 +1,129 @@
+import pyqtgraph as pg
+from PyQt5.QtCore import QDateTime
+from PyQt5.QtWidgets import (
+    QVBoxLayout,
+    QHBoxLayout,
+    QComboBox,
+    QDateTimeEdit,
+    QPushButton,
+    QLabel,
+    QCheckBox
+)
+from pydm import Display
+from typing import Dict
+
+from backend.backend_cavity import BackendCavity
+from frontend.cavity_widget import RED_FILL_COLOR, PURPLE_FILL_COLOR, DARK_GRAY_COLOR
+from lcls_tools.superconducting.sc_linac import Machine
+from lcls_tools.superconducting.sc_linac_utils import ALL_CRYOMODULES
+
+DISPLAY_MACHINE = Machine(cavity_class=BackendCavity)
+
+
+class FaultCounter(Display):
```

Comment:
I would rename this because we already have a FaultCounter in fault.py (or rename that one, but just not have a duplicated name)